spinn_front_end_common/interface/interface_functions/local_tdma_builder.py

```diff
@@ -150,6 +164,65 @@ def __call__(
                     machine_time_step, time_scale_factor, fraction_of_waiting)
                 app_vertex.set_initial_offset(initial_offset)
 
+    @staticmethod
+    def _auto_config_times(
+            app_machine_quantity, time_between_cores, fraction_of_sending,
+            time_scale_factor, machine_time_step, application_graph,
+            n_keys_map, machine_graph, fraction_of_waiting):
+
+        # initial offset in time for first core for beginning of tdma
+        initial_offset_point = int(math.ceil(
+            (machine_time_step * time_scale_factor) * fraction_of_waiting))
+
+        # find worst combo (NOTE: it might not exist as an actual app vertex,
+        # but least it'll cover all the things in between)
+        max_cores = 0
+        max_phases = 0
+        for app_vertex in application_graph.vertices:
+            if isinstance(app_vertex, TDMAAwareApplicationVertex):
+                cores = app_vertex.get_n_cores()
+                phases = app_vertex.find_n_phases_for(
+                    machine_graph, n_keys_map)
+                max_cores = max(max_cores, cores)
+                max_phases = max(max_phases, phases)
+        logger.info(
+            "max cores {} and max phases {}".format(max_cores, max_phases))
+
+        # overall time of the TDMA window minus initial offset
+        overall_time_available = (
+            ((time_scale_factor * machine_time_step) * fraction_of_sending) -
+            initial_offset_point)
+
+        # easier bool compares
+        core_set = time_between_cores is not None
+        app_set = app_machine_quantity is not None
+
+        # adjust time between cores to fit time scale
+        if not core_set and app_set:
+            n_slots = int(math.ceil(max_cores / app_machine_quantity))
+            time_per_phase = (
+                int(math.ceil(overall_time_available / max_phases)))
+            # NOTE the plus 1 ensures the last core finishes, if its the worst
+            # in terms of n keys to transmit
+            time_between_cores = time_per_phase / (n_slots + 1)
+            logger.info(
+                "adjusted time between cores is {}".format(time_between_cores))
+
+        # adjust cores at same time to fit time between cores.
+        if core_set and not app_set:
+            time_per_phase = (
+                int(math.ceil(overall_time_available / max_phases)))
+            max_slots = int(math.floor(time_per_phase / time_between_cores))
+            app_machine_quantity = int(math.ceil(max_cores / max_slots))
+            logger.info("adjusted app_machine_quantity is {}".format(
+                app_machine_quantity))
```

Comment:
More information and capital letter to start

Reply:
fixed and switched to debug